assets/js/directives/adverts.directive.js

```diff
@@ -20,7 +20,8 @@ function adverts (Adverts, $window) {
     Adverts.fetchOnce();
 
     scope.visit = (ad) => {
-      $window.open(ad.link, '_blank');
+      var url = $rootScope.apiDomain + 'ads/out?id=' + ad.id;
+      $window.open(url, '_blank');
```

Comment:
Can you use `ng-href` in the view instead of `$window.open()` and then add `noopener noreferrer`?